src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -898,7 +898,7 @@ protected virtual void ReportUnassigned(Symbol symbol, SyntaxNode node, int slot
 
             if (skipIfUseBeforeDeclaration &&
                 symbol.Kind == SymbolKind.Local &&
-                (symbol.Locations.Length == 0 || node.Span.End < symbol.Locations[0].SourceSpan.Start))
+                (symbol.Locations.Length == 0 || node.Span.End < symbol.Locations.FirstOrNone().SourceSpan.Start))
```

Comment:
>FirstOrNone() [](start = 82, length = 13)

Is this change necessary?

Reply:
I could revert it (ie. not necessary), but it is a good change to make. I ran into this as part of working on this PR.
Let me know what you think

---
In reply to: [355073423](https://github.com/dotnet/roslyn/pull/39370#discussion_r355073423) [](ancestors = 355073423)